### ISSUE TITLE
[Suggestion] Read Transfer Rate for Energy Interfaces

### ISSUE THREAD

**Issue by AgentA1cr** [CyclopsMC/IntegratedTunnels#90]
#### Issue type:
- :heavy_plus_sign: Feature request

It would be useful if the rf/t flowing through an energy interface could be written to a variable.

**Comment by josephcsible** [comment]
You can already easily get this with a Delayer. IMO, we already have too many "sugar" operators.

**Comment by AgentA1cr** [comment]
Thanks for the quick reply. I'll look into that method.

**Comment by rubensworks** [comment]
This can indeed be done with the delayer already.

**Comment by AgentA1cr** [comment]
Looking at the delayer, I can see how one could measure the net change in the energy level of a block over time, but not how one could measure raw energy throughput from one interface to another when the block  the energy is coming from is also receiving energy. I want to be able to connect the integrated dynamics system inline in my power network and measure the rf/t flowing through it. Is there a way to do this with the current implementation?

**Comment by josephcsible** [comment]
That's true, you can't currently do that at all. The problem is that in general, blocks don't keep track of that, so it's not something we can just add a reader for.

**Comment by AgentA1cr** [comment]
I don't want to measure it in an external block. I want to measure it in an Integrated Tunnels energy interface. I want to know how much power an interface is receiving or outputting.

**Comment by rubensworks** [comment]
This could be done by implementing a new type of energy interface, which tracks the energy that was moved per tick.

I'll reopen this issue, just in case anyone would be interested in implementing this in an addon mod. But this most likely won't be implemented in IT.

**Comment by josephcsible** [comment]
Here's an idea I have for something like this: a Flow Gauge Reader. It would have two sides that each implement IItemHandler, IFluidHandler, and IEnergyStorage. Whenever anything was pulled/pushed from one of those sides, it would proxy it to the other side, then take note of whatever it was. The other 4 sides would connect to ID networks and act like an aspect reader that shows whatever was transferred each way in the last tick.

**Comment by rubensworks** [comment]
@josephcsible That's also a possibility (perhaps even with 6 sides instead of only two).
But I still think this shouldn't be part of IT. This would make a nice addon though.

**Comment by josephcsible** [comment]
I actually thought it would fit well in ID, since it seems to be all about reading stuff.

**Comment by rubensworks** [comment]
Not sure,  it feels a bit too use-case-specific. I might change my mind in the future though.
